### PR TITLE
new: dev: Extending attribute injection.

### PR DESCRIPTION
**Pull request by ocelotl**
This change adds support for attribute injection in ports and links.
It also adds support for several links sharing the same attributes,
this is now possible:

```
[attribute=value] hs1:4 -- sw3:6 hs2:5 -- sw1:0
```

### CODE REVIEW

File: test/test_topology_injection.py [hunk 1]
```diff
@@ -271,16 +297,64 @@
 )
 
 
-def test_attribute_injection(tmpdir):
-    """
-    Test the configuration file is parsed correctly.
-    """
+@fixture
+def arranger(tmpdir):
```
Comment (dajose) [inline test/test_topology_injection.py]: why arranger?
also, please document this fixture

File: test/test_topology_injection.py [hunk 2]
```diff
@@ -271,16 +297,64 @@
...
+    # string. These SZN strings are the ones in the TOPOLOGY_MATCH_X constants.
+    # FIXME: Verify the next statement:
+    # The purpose of these subdirectories is to serve as different paths that
+    # can be used to test the file expansion feature of attribute injection.
     search_path = str(tmpdir.mkdir('test'))
     subfolder = str(tmpdir.mkdir('test/subfolder'))
 
+    # Filling the path placeholders in EXPECTED_PARSED_INJECTION_FILE, see the
+    # comment above that constant for more information.
+    expected = OrderedDict()
```
Comment (dajose) [inline test/test_topology_injection.py]: why not list comprehension here?

File: test/test_topology_injection.py [hunk 3]
```diff
@@ -293,37 +367,74 @@ def test_attribute_injection(tmpdir):
...
 
-        # Compare the actual and the expected
+        # Compare the actual and the expected ordered dictionaries, there
+        # should be no differences.
         differences = DeepDiff(actual, expected)
         assert not differences
 
-        # Test building with injected attributes
-        injected_attributes = actual[
+    finally:
```
Comment (dajose) [inline test/test_topology_injection.py]: you could add this to the fixture `finalizer`

File: lib/topology/injection.py [hunk 4]
```diff
@@ -52,7 +52,7 @@ def parse_attribute_injection(injection_file, search_paths=None):
     injection specifications. Each specification allows to list the files to
     modify and the modifiers to apply. Each modifier specify what entities
     (devices, ports or links) will be modified and which attributes will be
-    injected. The file support willcards and attributes matching.
+    injected. The file supports wildcards and attributes matching.
```
Comment (dajose) [inline lib/topology/injection.py]: wildcards don't work with ports and links, or do they?

**Comment by ocelotl** [comment]
To be superseded by #55.

**Pull request closed without merging**